software/tracksight/frontend/lib/widgetSchemas.ts

```diff
@@ -0,0 +1,14 @@
+import StateTimeline from "@/components/widgets/StateTimeline";
+
+const WIDGET_SCHEMAS = [
```

Comment:
a few notes on this
1. I would suggest against concrete variables which store type information. Keep type objects as types and concrete objects as concrete.
2. Instead of the current design, I would suggest using a typescript discriminated union to model the possible widgets that exist. This way, you can have true type checking without the runtime drawbacks